Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -336,9 +338,46 @@ public SecurityChanges ApplyUniverseSelection(Universe universe, DateTime dateTi
         /// </summary>
         /// <param name="utcStart">The current date time in utc</param>
         /// <returns>Will return true if any subscription was added</returns>
-        public bool AddPendingCurrencyDataFeeds(DateTime utcStart)
+        public bool AddPendingInternalDataFeeds(DateTime utcStart)
         {
             var added = false;
+            if (!_initializedSecurityBenchmark)
+            {
+                _initializedSecurityBenchmark = true;
+
+                var securityBenchmark = _algorithm.Benchmark as SecurityBenchmark;
+                if (securityBenchmark != null)
+                {
+                    // we want to start from the previous tradable bar so the benchmark security
+                    // never has 0 price
+                    var previousTradableBar = Time.GetStartTimeForTradeBars(
+                        securityBenchmark.Security.Exchange.Hours,
+                        utcStart.ConvertFromUtc(securityBenchmark.Security.Exchange.TimeZone),
+                        _algorithm.LiveMode ? Time.OneMinute : Time.OneDay,
+                        1,
+                        false).ConvertToUtc(securityBenchmark.Security.Exchange.TimeZone);
+
+                    var dataConfig = _algorithm.SubscriptionManager.SubscriptionDataConfigService.Add(
+                        securityBenchmark.Security.Symbol,
+                        _algorithm.LiveMode ? Resolution.Minute : Resolution.Daily,
+                        isInternalFeed: true,
+                        fillForward: false).First();
+
+                    if (dataConfig != null)
+                    {
+                        added |= _dataManager.AddSubscription(new SubscriptionRequest(
```

Comment:
I'm feeling better about this after additional thinking. The 'benchmark' is certainly an algorithm focused concern, so handling it explicitly in the algorithm layer is actually kind of nice 👍